code/weapon/weapons.cpp

```diff
@@ -8740,6 +8740,16 @@ float weapon_render_headon_bitmap(object* wep_objp, vec3d* headp, vec3d* tailp,
 	return side_alpha;
 }
 
+// renders another glow bitmap on top of the regular bitmap based on the angle of the camera to the front of the laser
+// uses the alpha data already gathered by weapon_render_headon_bitmap, because the alpha of the glow should always match the alpha of the laser
+void weapon_render_headon_glow_bitmap(float side_alpha, vec3d* headp, vec3d* tailp, int bitmap, float width1, float width2, int r, int g, int b){
+	float head_alpha = 1.0 - side_alpha;
+
+	r = (int)(r * head_alpha);   g = (int)(g * head_alpha);   b = (int)(b * head_alpha);
+
+	batching_add_laser(bitmap, headp, width1, tailp, width2, r, g, b);
+}
```

Comment:
This feels a bit trivial to have a function for. Yeah, it saves a bit of space for the caller, but I'm not sure it's offset by the introduction of this indirection.